src/main/java/net/mcreator/ui/action/impl/workspace/WorkspaceSettingsAction.java

```diff
@@ -199,6 +200,7 @@ public static void refactorWorkspace(MCreator mcreator, WorkspaceSettingsChange
 			} else { // in any other case, we need to regenerate the whole code
 				if (change.gradleCachesRebuildNeeded()) { // and rebuild caches when needed
 					ModAPIManager.deleteAPIs(mcreator.getWorkspace(), change.oldSettings);
+					ExternalTexture.invalidateCache(mcreator.getWorkspace());

```

Comment:
This is not ok as now when the generator is changed, caches will not be invalidated which is not correct behavior since old generatorconfiguration cache was abondoned

Reply:
At this point it looks like caching by `GeneratorConfiguration` becomes redundant, is that so?

Reply:
Yes, this is why you abandoned it. But your current solution is incomplete and will cause bugs too. You need more locations where you invalidate caches that you are missing right now (or more universally used location)

Reply:
The most common method to be called when workspace generator is touched should be `Generator#reloadGradleCaches()`, however this one performs its own operation and should probably not have to take care of texture caches. Or it's fine given how dependent external textures are on `ProjectJarManager`?

Reply:
Should be fine to call it there

Reply:
:heavy_check_mark: Should be fixed now